src/main/java/seedu/address/logic/commands/SetCommand.java

```diff
@@ -0,0 +1,142 @@
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.ingredient.Amount;
+import seedu.address.model.ingredient.Ingredient;
+import seedu.address.model.ingredient.IngredientName;
+
+/**
+ * Set the level of one specific ingredient to a specific level.
+ */
+public class SetCommand extends Command {
+
+    public static final String COMMAND_WORD = "set";
+
+    public static final String MESSAGE_ARGUMENTS = "Ingredient: %s, Amount: %2$s";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + " :set the ingredient in tCheck."
+            + "Parameters: "
+            + PREFIX_INGREDIENT
+            + PREFIX_AMOUNT
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_INGREDIENT + "Milk "
+            + PREFIX_AMOUNT + "90 ";
+
+    public static final String MESSAGE_SUCCESS = "Ingredient set: %1$s";
+    public static final String MESSAGE_NO_CHANGE = "Ingredient level already set.";
+    private final Ingredient target;
+    private final SetIngredientDescriptor setIngredientDescriptor;
+
+    /**
+     * Constructs a set command with the given ingredient name and amount.
+     */
+    public SetCommand(Ingredient target, SetIngredientDescriptor setIngredientDescriptor) {
+        requireNonNull(target);
+        requireNonNull(setIngredientDescriptor);
+
+        this.target = target;
+        this.setIngredientDescriptor = setIngredientDescriptor;
+
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Ingredient> lastShownList = model.getFilteredIngredientList();
+
+        Ingredient ingredientToEdit = lastShownList.get(lastShownList.indexOf(target));
+        Ingredient updatedIngredient = createSetIngredient(target, setIngredientDescriptor);
+
+        if (!ingredientToEdit.isSameIngredient(updatedIngredient)
+                && model.hasIngredient(updatedIngredient)) {
+            throw new CommandException(MESSAGE_NO_CHANGE);
+        }
+
+        model.setIngredient(ingredientToEdit, updatedIngredient);
+        return new CommandResult(String.format(MESSAGE_SUCCESS));
+    }
+
+    private static Ingredient createSetIngredient(Ingredient target,
+                                                  SetIngredientDescriptor setIngredientDescriptor) {
+        assert target != null;
+
+        Amount updatedAmount = setIngredientDescriptor.getAmount().orElse(target.getAmount());
+
+        return new Ingredient(target.getIngredientName(), updatedAmount);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+        if (!(other instanceof SetCommand)) {
+            return false;
+        }
+
+        SetCommand e = (SetCommand) other;
+
+        return target.equals(e.target)
+                && setIngredientDescriptor == e.setIngredientDescriptor;
```

Comment:
Should this be using `.equals` method instead to do the check?

Reply:
True, will change !